eclass/python-utils-r1.eclass

```diff
@@ -211,9 +211,12 @@ _python_impl_matches() {
 #
 # distutils-r1: Set within any of the python sub-phase functions.
 #
+# This will point to an automatically generated wrapper rather than the
+# usual path under /usr/bin in order to accomodate cross-compiling.
+#
 # Example value:
 # @CODE
-# /usr/bin/python2.7
+# /var/tmp/portage/dev-python/pyblake2-1.2.3/temp/python2.7/bin/python2.7
```

Comment:
I think this will break stuff. One of the points of PYTHON being the real path is that people actually hardcode the path given. If you change it, then I think we'll end up with having scripts referencing `#!/var/tmp/portage/dev-python/pyblake2-1.2.3/temp/python2.7/bin/python2.7` installed.

Reply:
It's true that I did not consider this but I believe our python-exec stuff prevents this from happening with shebangs. I grepped my entire test cross system and did not find a single instance of a path like that in any non-binary file. I suppose it could hypothetically appear in other places but I'll build the whole of dev-python to find them if that's what it takes. Unless you have a better idea? Having PYTHON point to the wrapper is not necessary for most packages, generally just the ones that don't use distutils, so we could tackle those individually but it would still be quite tedious.

Reply:
It's not `dev-python/` I'm worried about, it's everything outside `dev-python/`. Especially cheapo build systems.

Reply:
python-exec doesn't affect the shebangs in scripts themselves. They're preserved as-is and respected, in `/usr/lib/python-exec/*`.

Reply:
(this makes me think of adding a QA check to Portage to make sure thingies don't install stuff with generic shebangs there)

Reply:
Oh, and please test it on regular system as well ;-). I'm not really opposed to those changes (though I haven't reviewed them completely) but I'm afraid this is going to cause silent retroactive breakage of stuff.